numba/tests/test_parallel_backend.py

```diff
@@ -862,33 +862,32 @@ class TestInitSafetyIssues(TestCase):
 
     _DEBUG = False
 
+    def run_cmd(self, cmdline):
```

Comment:
I just looked at the documentation for `subprocess` for Python 3.6 (the minimal version supported by Numba) and it turns out that the `run` function is equipped with a `timeout` argument. I believe the construct below using `Popen` was written for an older version of Python and the `subprocess` module. Perhaps `run_cmd` can now be replaced with a more modern variant using the `run` function and the `timeout` argument? I think it would shorten the code significantly.

Reply:
Good point, am going to defer to https://github.com/numba/numba/issues/5989 though if that's ok as we know this pattern works for our, often strange, use cases.

Reply:
Sounds good!